move/gas_service/sources/gas_service.move

```diff
@@ -71,6 +81,34 @@ fun init(ctx: &mut TxContext) {
     );
 }
 
+// ------
+// Macros
+// ------
+// We allow unused functions here, vbecause if in the future anyone wants to access the balance mutliple times
+// they should load it with data first. The getter `balance!()` is here for tests and completeness.
+#[allow(unused_function)]
+macro fun data($self: &GasService): &GasServiceDataV0 {
+    let gas_service = $self;
+    gas_service.inner.load_value<GasServiceDataV0>()
+}
+
+#[allow(unused_function)]
+macro fun data_mut($self: &GasService): &mut GasServiceDataV0 {
+    let gas_service = $self;
+    gas_service.inner.load_value_mut<GasServiceDataV0>()
+}
+
+#[allow(unused_function)]
+macro fun balance($self: &GasService): &Balance<SUI> {
+    let gas_service = $self;
+    gas_service.inner.load_value<GasServiceDataV0>().balance()
+}
+
+macro fun balance_mut($self: &GasService): &mut Balance<SUI> {
```

Comment:
I prefer to avoid defining a macro for each method on the underlying object wherever possible and use `self.data!().xyz()` instead. we can see how this evolves, just don't want it to get out of hand